static/panes/compiler.js

```diff
@@ -560,6 +632,24 @@ Compiler.prototype.setAssembly = function (asm) {
         this.outputEditor.revealLine(currentTopLine);
     }
 
+    var decorations = [];
```

Comment:
I think here you'll need to add a new entry in `this.decorations` and then call `this.updateDecorations()` -- else you'll remove existing decorations. Take a look at the other spots that use `this.decorations` and you'll see this pattern used elsewhere.

Sorry it's not more obvious!

Reply:
Thank you, I haven't seen this function. I updated the code to use this.